analyzers/src/SonarAnalyzer.CSharp/Rules/Utilities/TokenTypeAnalyzer.cs

```diff
@@ -63,9 +65,40 @@ private sealed class TokenClassifier : TokenClassifierBase
                     SyntaxKind.InterpolatedStringTextToken,
                     SyntaxKind.InterpolatedStringEndToken,
                     SyntaxKindEx.InterpolatedRawStringEndToken);
+
+            protected override TokenTypeInfo.Types.TokenInfo ClassifyIdentifier(SyntaxToken token) =>
+                token.Parent switch
+                {
+                    FromClauseSyntax => null,
+                    LetClauseSyntax => null,
+                    JoinClauseSyntax => null,
+                    JoinIntoClauseSyntax => null,
+                    QueryContinuationSyntax => null,
+                    VariableDeclaratorSyntax => null,
+                    LabeledStatementSyntax => null,
+                    ForEachStatementSyntax => null,
+                    CatchDeclarationSyntax => null,
+                    ExternAliasDirectiveSyntax => null,
+                    EnumMemberDeclarationSyntax => null,
+                    MethodDeclarationSyntax => null,
+                    PropertyDeclarationSyntax => null,
+                    EventDeclarationSyntax => null,
+                    AccessorDeclarationSyntax => null,
+                    ParameterSyntax => null,
+                    var x when FunctionPointerUnmanagedCallingConventionSyntaxWrapper.IsInstance(x) => null,
+                    var x when TupleElementSyntaxWrapper.IsInstance(x) => null,
+                    var x when LocalFunctionStatementSyntaxWrapper.IsInstance(x) => null,
+                    var x when SingleVariableDesignationSyntaxWrapper.IsInstance(x) => null,
+                    TypeParameterSyntax => TokenInfo(token, TokenType.TypeName),
+                    BaseTypeDeclarationSyntax => TokenInfo(token, TokenType.TypeName),
+                    DelegateDeclarationSyntax => TokenInfo(token, TokenType.TypeName),
+                    ConstructorDeclarationSyntax => TokenInfo(token, TokenType.TypeName),
+                    DestructorDeclarationSyntax => TokenInfo(token, TokenType.TypeName),
```

Comment:
For some reasons I missed [AttributeTargetSpecifierSyntax.Identifier](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.csharp.syntax.attributetargetspecifiersyntax.identifier?view=roslyn-dotnet-4.6.0#microsoft-codeanalysis-csharp-syntax-attributetargetspecifiersyntax-identifier).

Reply:
Might all be parsed as keywords. A list of valid attribute targets is 

* assembly
* module
* event
* field
* method
* param
* property
* return
* type
* typevar